pages/admin/publisher/[id].js

```diff
@@ -1,22 +1,42 @@
-import { ResourceEditor } from 'giftpub'
+import React from 'react'
+import { DatasetEditor } from 'giftpub'
+import Error from 'next/error'
+import { getSession } from 'next-auth/client'
+import Metadata from '../../../lib/Metadata'
 
-export default function Publisher({ lfs, organisationId, datasetId }) {
+export default function Publisher({ lfsServerUrl, dataset, organizationId }) {
+  
   const config = {
-    datasetId: datasetId,
-    api: 'http://127.0.0.1:5000',
-    lfs: lfs,
-    authToken: 'be270cae-1c77-4853-b8c1-30b6cf5e9228',
-    organisationId: organisationId,
-    resourceId: '',
+    dataset: dataset,
+    lfsServerUrl: lfsServerUrl,
+    authorizeApi: '/api/authorize/',
+    metastoreApi: '/api/dataset/',
+    organizationId: organizationId
   }
+
   // eslint-disable-next-line react/react-in-jsx-scope
-  return <ResourceEditor config={config} resource="" />
+  return (
+    <>
+      { (dataset) && (<DatasetEditor config={config}/>) }
+      { (!dataset) && (<Error statusCode={404}/>)}
+    </>
+  )
 }
 
-Publisher.getInitialProps = async (ctx) => {
+Publisher.getInitialProps = async (context) => {
+
+  const metadata = new Metadata()
+  const {req} = context
+  const session = await getSession({ req })
+  const { user } = session
+
+  const userToken = process.env.APP_GITHUB_KEY
+
+  const data = await metadata.fetchMetadata(context.query.id,user,userToken)
```

Comment:
So here, you pass the name of the repo to the fetch method. Remember name not ID. This returns the metadata plus the id which you can then use.

Reply:
`id` here is from `admin/publisher/id`